extensions/funqy/funqy-knative-events/deployment/src/test/java/io/quarkus/funqy/test/WithOverlapingAttributeFilter.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.funqy.test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.quarkus.funqy.Funq;
+import io.quarkus.funqy.knative.events.CloudEventMapping;
+import io.quarkus.funqy.knative.events.EventAttribute;
+
+public class WithOverlapingAttributeFilter {
+
```

Comment:
@mswiderski we don't want test whether there is an overlap.

We want to know whether one filter is super-set of another.

Example A:
```java
    @Funq
    @CloudEventMapping(trigger = "listOfStrings", attributes = {
            @EventAttribute(name = "source", value = "test"),
            @EventAttribute(name = "custom", value = "value") })
    public String foo(List<Identity> identityList) {
        return "";
    }

    @Funq
    @CloudEventMapping(trigger = "listOfStrings", attributes = {
            @EventAttribute(name = "source", value = "test"),
            @EventAttribute(name = "custom", value = "someOtherValue") })
    public String bar(List<Identity> identityList) {
        return "";
    }
```

This won't compile because there is overlap. However those filters are not in conflict.

Example B:
```java
    @Funq
    @CloudEventMapping(trigger = "listOfStrings")
    public String foo(List<Identity> identityList) {
        return "";
    }

    @Funq
    @CloudEventMapping(trigger = "listOfStrings", attributes = {
            @EventAttribute(name = "source", value = "test") })
    public String bar(List<Identity> identityList) {
        return "";
    }
```

This would compile since there is no overlap. However this should fail since the `foo()` matches all that is matched by `bar()` (and more).

Reply:
well actually checking if one list is super-set of the another is not quite solution:

Example C:
```java
    @Funq
    @CloudEventMapping(trigger = "listOfStrings", attributes = {
            @EventAttribute(name = "source", value = "test"),
            @EventAttribute(name = "customA", value = "value") })
    public String foo(List<Identity> identityList) {
        return "";
    }

    @Funq
    @CloudEventMapping(trigger = "listOfStrings", attributes = {
            @EventAttribute(name = "source", value = "test"),
            @EventAttribute(name = "customB", value = "value") })
    public String bar(List<Identity> identityList) {
        return "";
    }
```
here is no strict sub-set relation between the two filters, still they are conflicting:
there may be an CloudEvent that matches both.

Reply:
I think we should have test for A,B,C.

Reply:
These three examples are describing three situation:

A -- every CE that matches `foo` doesn't match filter `bar` and vice versa
there is never a conflict, IMHO this should compile

B -- every CE that matches filter `bar` also matches `foo`
there is always conflict, IMHO this shouldn't compile

C -- there may be CEs that matches both, one, or none of the two.
there is sometime conflict, I am not sure what to do with this. Maybe it can compile, but return `409` in runtime?